Create 04_detection.md for batdetect2/msds

aditya-uw opened the pull request:
- Added some initial details into the detection protocol .md file. 
- Tried to take "I" and "you" pronouns out of original response.

@leewujung or @ctuguinay: Feel free to suggesting additions or things to remove!

ctuguinay commented:
From Slack: Currently going through the workflow that you described in the md file and I hit an error, and the error is printing out a text file with this. I've tried changing this line to batdetect2.bat_detect.utils.detector_utils as du and similar variants, but still no progress in getting it to run. 
![image](https://github.com/uw-echospace/union-bay-bats/assets/87830138/b7d47bc1-cdc2-4148-b2db-4dbd8c36b2bc)

As Wu-Jung also said over Slack, maybe something could be added for pip installing batdetect2 instead of submoduling it.

ctuguinay commented:
This looks good to me! Is this updated for your most recent script for usable files?

aditya-uw commented:
@ctuguinay No it is not! I'll keep this open and add those details by Monday, I plan on refining that functionality by then!

aditya-uw commented:
Hi @ctuguinay, when you're free could you look over detection.md again and see if it's good? I added info about the usable files and other troubleshooting details but I might need further help organizing the sections. Let me know!

ctuguinay commented on protocol/04_detection.md:
```diff
@@ -0,0 +1,55 @@
+
+## Pre-steps:
+
+- Fork `aditya-uw/bat-detector-msds` to access the batdetect2-pipeline branch in that repository.
+  - This is where the bash script that runs the pipeline is stored.
+- Download the forked repository with `git clone (your forked repository)` on the Linux machine with the mounted UBNA hard drives.
+- Follow the forked repository's README.md instructions about `pip install -r requirements.txt` and updating the submodule.
+  - It is ideal to install requirements and dependencies in a virtual environment with python=3.x.
```
Maybe somewhere in here, you can write something about fetching and pulling changes you made in your own repository. This would be especially helpful in the case where someone else wants the bat call CSVs for certain deployments and you've already run the pipeline for them and updated your branch.

Also could you add something about switching to the batdetect2-pipeline?

ctuguinay commented on protocol/04_detection.md:
```diff
@@ -0,0 +1,55 @@
+   - 9:29:27 is the time the pipeline still needs to generate detections for all segments. (HH:MM:SS)
+   - 4.55s/it is the time it takes to generate detections per segment for our pipeline on our chosen machine.
+
+### Slight modifications to pipeline:
+Deleting segments after detections have been generated for all files recovered.
```
I'm not too sure how necessary this part is. This seems more for the developer or potential developer of the code and not the people interested in solely running the pipeline.

ctuguinay commented on protocol/04_detection.md:
```diff
@@ -0,0 +1,55 @@
+   - 1 is the number of segments whose detections have been generated.
+   - 7560 is the total number of segments that were generated by the pipeline.
+   - 0:00:05 is the time that has passed for generating detections. (HH:MM:SS)
+   - 9:29:27 is the time the pipeline still needs to generate detections for all segments. (HH:MM:SS)
+   - 4.55s/it is the time it takes to generate detections per segment for our pipeline on our chosen machine.
```
For the first value, could you include a range of time that you would expect this to be in.

Could you also add a note about what a good value for the bottom value looks like and what a "bad" value for it may look like. And could you add a note about what to do if this value is "bad"?

ctuguinay commented on protocol/04_detection.md:
```diff
@@ -0,0 +1,55 @@
+   - External microphone recordings of 30-min lengths may exist in our input directories but these recordings lack real world data.
+   - Other recordings that are 488-bytes may also exist in our input directories and lack real world data.
+   - We are still investigating why these "unusable" files are generated by the Audiomoth but our pipeline ignores them using os.stat() and exiftool.
+3) The pipeline will then display the number of usable files out of the total number of files in the input directory.
+4) The pipeline will then begin generating segments for all usable files. This may take some time but still is a fraction of the time it will take to generate detections.
```
In my running of the pipeline, it seems as though both 3 and 4 take some time, about 10 minutes or so. Did you experience something similar?

aditya-uw replied:
Yeah! I'll make a note about these durations

ctuguinay reviewed (commented):
Hey @aditya-uw, I left some comments and I'm still currently running your pipeline. I may update this later on based on what happens when the pipeline finishes. Awesome job at documenting this process!

aditya-uw commented:
Okay @ctuguinay! I tried addressing your comments! Could you look and let me know if I'm missing anything specific? Thanks and my b for not getting back sooner!

ctuguinay commented:
@aditya-uw This looks good to me, thank you for the changes!
